### PR TITLE
#11834 Use reversed instead of creating a copy of slotData for every call to _getSlotValue

### LINKED ISSUE #11834: twisted.web._flatten._getSlotValue copies slotData on every call.
**Describe the incorrect behavior you saw**

`_getSlotValue` attempts retrieve a value for a named slot from the `slotData` stack by using slice syntax to reverse it and then iterate over it.  

This also causes it create a copy of the list.

In templates with large amounts of data such those used by pydoctor's summary pages.  This copying ends up being a significant amount of time spent generating summary templates.

**Describe how to cause this behavior**

I discovered this issue while running pydoctor under the austin profiler.

I can reproduce it on twisted's own apidocs.

You can view the results of this profiling in speedscope: https://www.speedscope.app/#profileURL=https%3A%2F%2Fgist.githubusercontent.com%2Fdreid%2F197566471f39a96523f5065d19d0bf7f%2Fraw%2F3e7ec92a17bc82d40acceb1e2efcaa3ef7c8ef07%2Ftwisted-austin-trunk.speedscope

In the above profile `_getSlotValue` accounts for ~55s of the total runtime which was 509.42.

**Describe the correct behavior you'd like to see**
`slotData` should not need to be copied on every call to `_getSlotValue` simply to iterate over it in reverse.

Initial profiling using `reversed` instead resulted in reduced runtime to 357.36s

speedscope profile of using reversed is here: https://www.speedscope.app/#profileURL=https%3A%2F%2Fgist.githubusercontent.com%2Fdreid%2F197566471f39a96523f5065d19d0bf7f%2Fraw%2F3e7ec92a17bc82d40acceb1e2efcaa3ef7c8ef07%2Ftwisted-pydoctor-reversed.speedscope

**Testing environment**
 - Operating System and Version; paste the output of these commands:
   - on Linux, `uname -a ; cat /etc/lsb-release`
   - on Windows, `systeminfo | Findstr /i "OS"`
   - on macOS, `sw_vers`
 - Twisted version [e.g. 22.2.0]
   - please paste the output of `twist --version` and `pip --freeze`
 - Reactor [e.g. select, iocp]

**Additional context**

### PR DESCRIPTION
**Pull request by dreid**
## Scope and purpose

Fixes #11834

## Contributor Checklist:

This process applies to *all* pull requests - no matter how small.
Have a look at [our developer documentation](https://docs.twisted.org/en/latest/core/development/dev-process.html) before submitting your Pull Request.

Below is a non-exhaustive list (as a reminder):

* The title of the PR should describe the changes and starts with the associated issue number, like “#1234 Brief description”.
* A release notes news fragment file was create in src/twisted/newsfragments/ (see: [Release notes fragments docs.](https://docs.twisted.org/en/latest/core/development/dev-process.html#release-notes-management))
* The automated tests were updated.
* Once all checks are green, request a review by leaving a comment that contains exactly the string `please review`.
  Our bot will trigger the review process, by applying the pending review label
  and requesting a review from the Twisted dev team.

### CODE REVIEW

**Comment by dreid** [comment]
I do not understand the pypy failures.  It seems unrelated but also it seems to fail consistently.  I'm I couldn't find any other examples of similar failures on outstanding PRs but I only checked 3 or 4.

File: src/twisted/newsfragments/11834.bugfix [hunk 1]
```diff
@@ -0,0 +1 @@
+Avoid copying slotData on every call to _getSlotValue.
```
Comment (exarkun) [inline src/twisted/newsfragments/11834.bugfix]:
```suggestion
`twisted.web.template` now avoids unnecessary copying and is faster, particularly for templates with deep nesting.
```

**Comment by exarkun** [comment]
> I do not understand the pypy failures. It seems unrelated but also it seems to fail consistently. I'm I couldn't find any other examples of similar failures on outstanding PRs but I only checked 3 or 4.

The most recent trunk run has exactly these failures.

**Review by exarkun — APPROVED** [review]
I couldn't get the speedscope URL to load which I'm a bit sad about.

I also made a suggestion for a change to the news fragment.

The CI failures look pre-existing to me.

Please adjust the news fragment and then merge.

Thanks.

**Comment by dreid** [comment]
@exarkun Thank you for the review, bummer about the speedscope link.  I've updated the news fragment.  I do not have permissions to merge PRs.

**Pull request merged**